### ISSUE TITLE
Data leakage

### ISSUE THREAD

**Issue by romanceyu** [ArnabKumarRoy02/ResEmoteNet#42]
Hello! First of all, thank you very much for your work in experiment, which has helped me a lot in understanding this field.

After carefully reading the Fer2013 dataset you provided, We noticed that some images appeared in the training dataset, test set, and validation set simultaneously, which could lead to data leakage during model evaluation, thus affecting the objectivity of the results.

We would like to confirm with you whether the same image may be intentionally or unintentionally assigned to multiple datasets during the data partitioning process. If this is the case, it may be necessary to further discuss how to better split the data and ensure the mutual exclusion of the training dataset, validation set, and test set in order to more accurately measure the generalization performance of the model.

If you have time or more details to share, we look forward to your reply and further communication.

![Image](https://github.com/user-attachments/assets/68c00c8a-3996-4e6b-8dfb-7f0734593500)

![Image](https://github.com/user-attachments/assets/f8b77536-81ad-42b9-bd4d-57e78cd6a045)

![Image](https://github.com/user-attachments/assets/e9266594-483e-4a5b-b8f1-f5f04495423c)

![Image](https://github.com/user-attachments/assets/4b3192f6-3cdb-48af-b8c8-6222b37cd2b9)

![Image](https://github.com/user-attachments/assets/4dde1f1c-3c53-4db1-946c-6e86d17fbb9c)

**Comment by wjy110** [comment]
It is difficult to achieve the accuracy in the paper without the author's dataset

**Comment by cyinen** [comment]
看来作者在训练集里面放了测试集的数据

**Comment by BITLiuYu** [comment]
This is a serious case of academic misconduct, and I believe it should be reported.

**Comment by romanceyu** [comment]
> 看来作者在训练集里面放了测试集的数据

大佬，能和你交流一下吗？我和我的朋友想和你交流一下

**Comment by romanceyu** [comment]
> > 看来作者在训练集里面放了测试集的数据
> 
> 大佬，能和你交流一下吗？我和我的朋友想和你交流一下(或者q群：1034544347)

**Comment by ArnabKumarRoy02** [comment]
Thanks @romanceyu for pointing this out and others for supporting! I really appreciate the detailed review and for catching this issue.

You’re absolutely right, this data leakage between the training and testing sets is a significant problem, and it needs to be fixed. I’ll be looking into this and updating the dataset accordingly.

I’ll post an update here once the issue is resolved. Thanks again for your valuable feedback!

**Comment by ArnabKumarRoy02** [comment]
Hey @romanceyu, @BITLiuYu, @wjy110 and @chenyin2, I would like to show the images that you guys found to be same or similar in the FER2013 dataset which was uploaded to Google Drive (now claiming to an academic misconduct) amongst different sets: Training, Validation and Testing. The same issue occurs in the original dataset as well.

This [link](https://www.kaggle.com/datasets/deadskull7/fer2013) is the original FER2013 dataset which I even confirmed with @mkonuk66 who helped me with the following study.

 - Rather than looking for the same exact image, me and @mkonuk66 used PHash for similarity prediction. Here's the script we used, you can do the same from your ends.
```python
import os
import imagehash
from PIL import Image
from collections import defaultdict

def compute_phash(image_path):
    """Compute the Perceptual Hash (PHash) of an image."""
    try:
        with Image.open(image_path) as img:
            return imagehash.phash(img)
    except Exception as e:
        print(f"Error processing {image_path}: {e}")
        return None

def get_image_hashes(directory):
    """Get a dictionary of image hashes for a given directory."""
    image_hashes = defaultdict(list)
    for root, _, files in os.walk(directory):
        for file in files:
            if file.lower().endswith(('.png', '.jpg', '.jpeg')):
                file_path = os.path.join(root, file)
                img_hash = compute_phash(file_path)
                if img_hash is not None:
                    image_hashes[img_hash].append(file_path)
    return image_hashes

def find_similar_images(train_dir, test_dir, threshold=5):
    """
    Find visually similar images between training and testing sets using PHash.
    The threshold determines similarity (lower is stricter, higher allows more variation).
    """
    print("Computing PHash for training set...")
    train_hashes = get_image_hashes(train_dir)
    
    print("Computing PHash for testing set...")
    test_hashes = get_image_hashes(test_dir)

    duplicate_pairs = []  # Stores matching (train_image, test_image) pairs

    print("\nChecking for similar images...")
    for test_hash, test_files in test_hashes.items():
        for train_hash, train_files in train_hashes.items():
            if test_hash - train_hash <= threshold:  # Check Hamming distance
                for test_file in test_files:
                    for train_file in train_files:
                        duplicate_pairs.append((train_file, test_file))
                        print(f"Duplicate found:\n - Train: {train_file}\n - Test:  {test_file}\n")

    # Save results
    if duplicate_pairs:
        with open("duplicates.txt", "w") as f:
            for train_img, test_img in duplicate_pairs:
                f.write(f"Train: {train_img} | Test: {test_img}\n")
        print(f"\n{len(duplicate_pairs)} duplicate(s) found. List saved in 'duplicates.txt'.")
    else:
        print("\nNo duplicates found!")

# Update paths to your dataset
train_dir = "train_folder"
test_dir = "test_folder"

find_similar_images(train_dir, test_dir, threshold=5)

```

The above script looks for similar images that were found in the original dataset and lists them out in the txt file. We performed this on the original FER2013 dataset and found out that there are `771` similar images between the Training and PrivateTest (Test set) and `661` similar images between Training and PublicTest (Validation set) with `threshold=5`.

Following are some examples:

1. For Class 0

Found in Training set.
![Image](https://github.com/user-attachments/assets/23f6d2c4-d770-40a0-8d3b-ee361c09c3bc)

Found in Test set.
![Image](https://github.com/user-attachments/assets/12c3afd7-0180-4348-b70a-672ba0eb516d)

2. For Class 1

Found in Training set.
![Image](https://github.com/user-attachments/assets/8ec57a16-93b4-4c34-b8ef-aa2d438ad952)

Found in Test set.
![Image](https://github.com/user-attachments/assets/3ec3780c-ec60-4ecf-a01b-7691c7fd9d93)

These above images are just a few examples of similar images found in the original dataset, there are many more. You can even increase the value of `threshold` for finding even more similarity. 

In my version of the FER2013 dataset, I had to use augmentation in order to increase more samples thereby getting a better performance. Hence there are even more similarity.

Before making accusations against me in this case, I urge you to review the original dataset.

**Comment by wjy110** [comment]
May I ask if you used the same dataset in your paper for performance comparison with other models, which is the dataset you expanded through data augmentation? If you used the same dataset, then your model undoubtedly achieved the most advanced performance. If you only referenced publicly available data from other model papers, such as EmoNet's performance of 76.12 in your paper, was this data obtained by training your enhanced dataset? If not, did you consider the fundamental principle of controlling variables in scientific experiments, so as to verify that your model is superior to other models.
One more thing, I would like to ask if the focus of your paper is on data augmentation or innovative proposal of a new model that achieves SOTA. If it is data augmentation, do you want to demonstrate that other models have improved through training on your dataset? If your work focuses on the new model you proposed, do you want to compare the accuracy of each model while ensuring the same dataset.
I am new to deep learning models and may not have a deep understanding of the rules and standards for evaluating deep learning models. Some of the questions I have raised may be low-level or incorrect. Please forgive me.

**Comment by wjy110** [comment]
I would like to ask whether the performance comparison in your paper is conducted using the same dataset for all models. Specifically, is your model trained and evaluated on a dataset that has been expanded through data augmentation? If all models in the comparison are trained on the same dataset, then your model has undoubtedly achieved state-of-the-art (SOTA) performance.

However, if you are merely citing performance metrics from other papers—for example, if the reported accuracy of EmoNet in your paper is 76.12%—was this result obtained by retraining the model on your augmented dataset? If not, have you considered the fundamental scientific principle of controlling variables? Ensuring a fair comparison under identical experimental conditions is crucial to objectively validating the superiority of your model over others.

Additionally, I would like to clarify the main focus of your research. Is it primarily about the contribution of data augmentation, or have you proposed an innovative model that achieves SOTA performance? If your work focuses on data augmentation, would it be beneficial to show whether other models also improve when trained on your dataset? On the other hand, if your primary contribution is a novel model, should the comparison be conducted under identical dataset conditions to ensure fairness?

As I am new to deep learning models, my understanding of evaluation standards and best practices may be limited. Some of my questions might be basic or incorrect, and I sincerely appreciate your patience and guidance.
